### PR TITLE
Fix Android black box test connection locally

### PR DESCRIPTION
**Pull request by johnxwork**

### CODE REVIEW

File: cobalt/black_box_tests/tests/allow_eval.py [hunk 1]
```diff
@@ -25,9 +25,9 @@ class AllowEvalTest(black_box_tests.BlackBoxTestCase):
   """Ensure that client page can use eval() when CSP is missing."""
 
   def test_allow_eval(self):
-
     with ThreadedWebServer(binding_address=self.GetBindingAddress()) as server:
       url = server.GetURL(file_name='testdata/allow_eval.html')
 
-      with self.CreateCobaltRunner(url=url) as runner:
+      with self.CreateCobaltRunner(
+          url=url, web_server_port=server.GetPort()) as runner:
```
Comment (kaidokert) [inline cobalt/black_box_tests/tests/allow_eval.py:32]: passing `web_server_port` here requires modifying _all_ files. Is there perhaps another way how this argument can be passed down to the runner ? It already carries `launcher_params` and `target_params`

File: cobalt/black_box_tests/black_box_tests.py [hunk 2]
```diff
@@ -155,7 +158,9 @@ def CreateCobaltRunner(self, url=None, target_params=None, **kwargs):
         launcher_params=_launcher_params,
         url=url,
         target_params=all_target_params,
-        **kwargs)
+        web_server_port=web_server_port,
```
Comment (kaidokert) [inline cobalt/black_box_tests/black_box_tests.py:161]: Would it be possible to move the `web_server_port` to be a member of `_launcher_params` so it gets tunneled trough to the runner, without having to modify all callers ?

Comment (johnxwork) [reply cobalt/black_box_tests/black_box_tests.py:161]: After moving web_server_port into target_params, we will still need to modify all the tests to pass in target_params.

Also, target_params is used to pass in Cobalt flags such as `--preload`. It feels strange to pass in variables for the launcher.

File: cobalt/black_box_tests/tests/allow_eval.py (hunk 1 above)
Comment (johnxwork) [reply cobalt/black_box_tests/tests/allow_eval.py:32]: The only option I see here is to extract the web server's creation and destruction from each test level to the test suite's initialization. I think that might be a good change. But we will definitely want a dedicated PR for such big modification.

**Review by oxve — CHANGES_REQUESTED** [review]
This PR includes a ton of formatting which makes it very hard to review. I'm not asking that you break out the formatting in this PR but to keep it in mind in the future.

Also, what is your formatter set to do? This looks like more than the standard pylint hook we run.

**Comment by johnxwork** [comment]
> This PR includes a ton of formatting which makes it very hard to review. I'm not asking that you break out the formatting in this PR but to keep it in mind in the future.
> 
> Also, what is your formatter set to do? This looks like more than the standard pylint hook we run.

I got the pre-commit hook wrong, so I used pyformat at some point. After that it became really difficult to revert :(

File: cobalt/black_box_tests/threaded_web_server.py [hunk 3]
```diff
@@ -66,16 +67,11 @@ def MakeCustomHeaderRequestHandlerClass(base_path, paths_to_headers):
...
-    dicts. Key is expected to be a substring of a file being served.
-
-    E.g. if you have a test with files foo.html and foo.js, you can serve them
-    separate headers with the following:
-    paths_to_headers = {
-      'foo.html': {
-        'Content-Security-Policy', "script-src 'unsafe-inline';"
-      },
-      'foo.js': {'Content-Security-Policy', "default-src 'self';"}
-    }
```
Comment (oxve) [inline cobalt/black_box_tests/threaded_web_server.py:78]: Looks like the formatting of this comment got messed up. Can you fix?

File: cobalt/black_box_tests/black_box_cobalt_runner.py [hunk 4]
```diff
@@ -43,14 +43,17 @@ class BlackBoxCobaltRunner(cobalt_runner.CobaltRunner):
...
+  def __init__(
+      self,
+      launcher_params,
+      url,
+      log_file=None,
+      target_params=None,
+      success_message=None,
+      poll_until_wait_seconds=POLL_UNTIL_WAIT_SECONDS,
+      web_server_port=None,
+      **kwargs,
```
Comment (oxve) [inline cobalt/black_box_tests/black_box_cobalt_runner.py:55]: Did you mean to leave this trailing comma?

File: cobalt/black_box_tests/black_box_cobalt_runner.py [hunk 5]
```diff
@@ -65,7 +68,9 @@ def __init__(self,
         target_params,
         success_message,
         poll_until_wait_seconds=poll_until_wait_seconds,
-        **kwargs)
+        web_server_port=web_server_port,
+        **kwargs,
```
Comment (oxve) [inline cobalt/black_box_tests/black_box_cobalt_runner.py:72]: Same here

File: cobalt/black_box_tests/black_box_cobalt_runner.py [hunk 6]
```diff
@@ -82,10 +87,12 @@ def PollUntilFoundOrTestsFailedWithReconnects(self, css_selector):
...
         is_failed = self.JSTestsFailed()
-      except (cobalt_runner.CobaltRunner.AssertException,
-              selenium_exceptions.NoSuchElementException,
-              selenium_exceptions.NoSuchWindowException,
-              selenium_exceptions.WebDriverException) as e:
+      except (
+          cobalt_runner.CobaltRunner.AssertException,
+          selenium_exceptions.NoSuchElementException,
+          selenium_exceptions.NoSuchWindowException,
+          selenium_exceptions.WebDriverException,
```
Comment (oxve) [inline cobalt/black_box_tests/black_box_cobalt_runner.py:94]: And here

File: cobalt/black_box_tests/proxy_server.py
Comment (oxve) [inline cobalt/black_box_tests/proxy_server.py:1]: More trailing commas in here

File: cobalt/tools/automated_testing/cobalt_runner.py [hunk 7]
```diff
@@ -272,7 +290,11 @@ def Run(self):
         loader_platform=self.launcher_params.loader_platform,
         loader_config=self.launcher_params.loader_config,
         loader_out_directory=self.launcher_params.loader_out_directory,
+        launcher_args=launcher_args,
         **self.kwargs)
+    #Start the Device->Host tunnel now in case Cobalt connects the host.
```
Comment (oxve) [inline cobalt/tools/automated_testing/cobalt_runner.py:295]: Missing a space between `#` and `Start`

File: cobalt/black_box_tests/black_box_cobalt_runner.py [hunk 8]
```diff
@@ -85,7 +87,7 @@ def PollUntilFoundOrTestsFailedWithReconnects(self, css_selector):
       except (cobalt_runner.CobaltRunner.AssertException,
               selenium_exceptions.NoSuchElementException,
               selenium_exceptions.NoSuchWindowException,
-              selenium_exceptions.WebDriverException) as e:
+              selenium_exceptions.WebDriverExceptin) as e:
```
Comment (oxve) [inline cobalt/black_box_tests/black_box_cobalt_runner.py:90]: Is this intentional?

File: cobalt/black_box_tests/tests/preload_font.py [hunk 9]
```diff
@@ -34,14 +34,16 @@ def test_preload_font(self):
...
+          web_server_port=server.GetPort()) as runner:
         runner.WaitForJSTestsSetup()
         runner.SendResume()
         start_time = time.time()
         while runner.IsInPreload():
           if time.time() - start_time > _MAX_RESUME_WAIT_SECONDS:
-            raise Exception('Cobalt can not exit preload mode after receiving'
-                            'resume signal')
+            raise ValueError('Cobalt can not exit preload mode after receiving'
+                             'resume signal')
```
Comment (oxve) [inline cobalt/black_box_tests/tests/preload_font.py:46]: Missing space between `receiving` and `resume`.

Not your fault but you might as well fix it here.

File: cobalt/black_box_tests/threaded_web_server.py [hunk 10]
```diff
@@ -145,9 +141,11 @@ def get_content(self):
 class ThreadedWebServer(object):
   """A HTTP WebServer that serves requests in a separate thread."""
 
-  def __init__(self,
-               handler=MakeRequestHandlerClass(os.path.dirname(__file__)),
-               binding_address=None):
+  def __init__(
+      self,
+      handler=MakeRequestHandlerClass(os.path.dirname(__file__)),
+      binding_address=None,
```
Comment (oxve) [inline cobalt/black_box_tests/threaded_web_server.py:147]: Another trailing comma here.

**Review by oxve — APPROVED** [review]
Just a couple of small items left. PTAL and fix.

Otherwise I'm approving

File: cobalt/black_box_tests/black_box_cobalt_runner.py [hunk 11]
```diff
@@ -103,17 +105,18 @@ def JSTestsSucceeded(self):
     # JavaScript test logic completion.
     self.PollUntilFound('[' + _TEST_STATUS_ELEMENT_NAME + ']')
     body_element = self.UniqueFind('body')
-    return body_element.get_attribute(
-        _TEST_STATUS_ELEMENT_NAME) == _JS_TEST_SUCCESS_MESSAGE
+    return (body_element.get_attribute(_TEST_STATUS_ELEMENT_NAME) ==
+            _JS_TEST_SUCCESS_MESSAGE)
```
Comment (TyHolc) [inline cobalt/black_box_tests/black_box_cobalt_runner.py:109]: Avoid wrapping return statements in parens if not necessary: go/pystyle#parentheses

File: cobalt/black_box_tests/black_box_cobalt_runner.py [hunk 12]
```diff
@@ -103,17 +105,18 @@ def JSTestsSucceeded(self):
...
     """Check failed test assertion in HTML page."""
 
     # Call onTestEnd() in black_box_js_test_utils.js to unblock the waiting for
     # JavaScript test logic completion.
     body_element = self.UniqueFind('body')
-    return body_element and body_element.get_attribute(
-        _TEST_STATUS_ELEMENT_NAME) == _JS_TEST_FAILURE_MESSAGE
+    return (body_element and
+            body_element.get_attribute(_TEST_STATUS_ELEMENT_NAME)
+            == _JS_TEST_FAILURE_MESSAGE)
```
Comment (TyHolc) [inline cobalt/black_box_tests/black_box_cobalt_runner.py:119]: Same here: go/pystyle#parentheses

File: starboard/tools/abstract_launcher.py [hunk 13]
```diff
@@ -283,6 +283,22 @@ def GetHostAndPortGivenPort(self, port):
...
+      host_port: The host_port to receive requests from the device.
+      device_port: The port on device to proxy on-device requests.
+
+    Returns:
+      True if succeed and false otherwise.
+    """
+    del host_port
+    del device_port
+    return False
+
```
Comment (TyHolc) [inline starboard/tools/abstract_launcher.py:301]: Do we want to include the matching RemoveDeviceToHostTunnel?

**Review by TyHolc — APPROVED** [review]

**Comment by kaidokert** [comment]
I'll re-state again - i'm not really wanting that extensive change in all tests, there _should_ be a less intrusive way to pipe the argument through without causing changes in _all_ tests. Especially because the arg is specific to a single platform only.

**Comment by johnxwork** [comment]
> wanting

Just removed all changes in python tests and turned to use a global variable for sending the port number.

**Review by kaidokert — APPROVED** [review]
This looks pretty clean to me - as far as this framework goes in any case. 

If the tests run, lgtm

**Pull request merged**